java/TrickyInputTest.java

```diff
@@ -0,0 +1,283 @@
+
+//See https://www.baeldung.com/hamcrest-text-matchers
+import static org.hamcrest.text.IsBlankString.blankOrNullString;
+
+import java.util.Map;
+
+//See https://junit.org/junit4/javadoc/latest/deprecated-list.html
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * This class contains a few test cases that work of the example provided in the assignment
+ * description.
+ * 
+ * @author stocker4141
+ *
+ */
+public class TrickyInputTest {
+    
+    
+    public final String boatFieldTop = "\u2554\u2550\u2550\u2550\u2550\u2550\u2550\u2550\u2550";
+    public final String boatFieldBottom = "\u255A\u2550\u2550\u2550\u2550\u2550\u2550\u2550\u2550";
+    public final String oceanFieldTop = "\u2564\u2550\u2550\u2550\u2550\u2550\u2550\u2550\u2557";
+    public final String oceanFieldBottom = "\u2567\u2550\u2550\u2550\u2550\u2550\u2550\u2550\u255D";
+    public final String tileTop = "\u2564\u2550\u2550\u2550";
+    public final String tileBottom = "\u2567\u2550\u2550\u2550";
+    
+
+    @BeforeClass
+    public static void checkPrecondition() {
+        /*
+         * Check only once that all the required variables and configurations are
+         * correctly set
+         */
+        PSTestUtils.validateTheExecutionEnvironment();
+    }
+    
+    private void assertFinalBoardAndMessageMatches(String[] expectedBoard, String[] expectedMessage, String stdOut) {
```

Comment:
It would be really **nice** if you could wrap the assertion into an Hamcrest text-matcher [1], so it would be easy to reuse across all the test cases.  Maybe you can look at [2][3]

[1] https://www.baeldung.com/hamcrest-text-matchers
[2] https://www.baeldung.com/hamcrest-custom-matchers

Reply:
Just so I am clear on this: Is the goal is to create an Hamcrest Matcher such that one can write:
`assertThat(actualBoard, matchesBoard(expectedBoard))`?

As far as checking if a fish is in a certain tile goes, I am unsure how this can be done. 
However, the ocean tile and the boat tile should be possible by checking the start / end of the lines that make up the board. Using a custom matcher possible calls might look like
`assertThat(actualBoard, is(capturedFish(3))` or 
`assertThat(actualBoard, is(saveFish(3))`.
Alternatively, a check like
`assertThat(actualBoard[<index>], endsWith(<oceanStringWithFish>))` 
could be wrapped in some static convenience method.
Is that what you had in mind and which alternative do you prefer?